ci/cron/src/Main.hs

```diff
@@ -183,12 +185,19 @@ fetch_gh_paginated url = do
                 (_, _, _, [url, rel]) -> (rel, url)
                 _ -> fail $ "Assumption violated: link header entry did not match regex.\nEntry: " <> l
 
-data PreVersion = PreVersion { prerelease :: Bool, tag :: Version }
-instance JSON.FromJSON PreVersion where
-    parseJSON = JSON.withObject "PreVersion" $ \v -> PreVersion
-        <$> v JSON..: "prerelease"
-        <*> let json_text = v JSON..: "tag_name"
-            in (version . Text.tail <$> json_text)
+data Asset = Asset { uri :: Network.URI.URI }
+instance JSON.FromJSON Asset where
+    parseJSON = JSON.withObject "Asset" $ \v -> Asset
+        <$> (do
+            url_as_string <- v JSON..: "browser_download_url"
+            return $ Data.Maybe.fromJust $ Network.URI.parseURI url_as_string)
```

Comment:
```suggestion
            Just url <- Network.URI.parseUri <$> v JSON..: "browser_download_url"
            return url)
```
This will make the parser fail rather than crash the whole program on a parse error.

Reply:
Neat! I do want the program to crash here, though. I _believe_ the parser failing will achieve that, and with any luck we'll get a better error message than from `fromJust`. Also saves me one import.

Reply:
Right, the program will still crash but the parser itself won’t which means you could compose it with other parsers, potentially get better errors, …. All not super important here but given that it’s not any longer and seems a bit cleaner, I don’t see a reason not to do it either.

Reply:
Oh yes it's done. Sorry if that was not clear, I was actually listing arguments in favour. 🙂